osf/models/mixins.py

```diff
@@ -687,9 +687,12 @@ class Meta:
 
     @cached_property
     def subject_hierarchy(self):
-        return [
-            s.object_hierarchy for s in self.subjects.exclude(children__in=self.subjects.all())
-        ]
+        subjects = self.subjects.all()
+        if subjects.count():
```

Comment:
Minor: It is slightly more performant to do `self.subjects.exists()` than to get the count.

Reply:
I was wondering about that, so this is true even if I've done self.subjects.all() in the previous line?

Reply:
Made this change when I merged.

Reply:
@pattisdr Yeah, `self.subjects.all()` returns a queryset, but the SQL doesn't get executed until you run `.count()`.